src/clients/lan_auto.ycp

```diff
@@ -62,6 +62,65 @@ map UpcaseCondSet (map ret, map m, string key) {
     return ret;
 }
 
+/**
+ * Merges two devices map into one.
+ *
+ * Maps are expected in NetworkInterfaces format. That is
+ * $[
+ *   type1:
+ *     $[
+ *       dev_name_1: $[ ... ],
+ *       ...
+ *      ],
+ *    ...
+ *  ]
+ *
+ * If a device definition is present in both maps, then the one from devices2
+ * wins.
+ *
+ * @param in_devs1 first map of devices in NetworkInterfaces format
+ * @param in_devs2 second map of devices in NetworkInterfaces format
+ *
+ * @return merged device map in NetworkInterfaces format or empty map
+ */
+define map MergeDevices(map<string, any> in_devs1, map<string, any> in_devs2)
+{
+    map merged_devices = $[];
+
+    y2milestone("MergeDevices: start (in_devs1: %1, in_devs2: %2)", in_devs1, in_devs2);
+
+    if(in_devs1 == nil || in_devs1 == {})
+    {
+        merged_devices = in_devs2 == nil ? $[] : in_devs2;
+        return merged_devices;
+    }
+    if(in_devs2 == nil || in_devs2 == {})
+    {
+        merged_devices = in_devs1 == nil ? $[] : in_devs1;
+        return merged_devices;
+    }
+
+    merged_devices = in_devs1;
+
+    y2milestone("MergeDevices: going to do real merge");
+
+    foreach(string type, any devices, in_devs2,
+    {
+        if(haskey(in_devs1, type))
+        {
+            map merged = union((map) in_devs1[type]:{}, (map) in_devs2[type]:{});
+            merged_devices = add(merged_devices, type, merged);
```

Comment:
honestly I don't know. May be I'm so used to ```Builtins.add``` from ruby branch that I've used that even here ;-)

Will fix it.